staging/src/k8s.io/kubectl/pkg/cmd/debug/profiles.go

```diff
@@ -130,10 +157,34 @@ func getDebugStyle(pod *corev1.Pod, target runtime.Object) (debugStyle, error) {
 	return unsupported, fmt.Errorf("objects of type %T are not supported", target)
 }
 
+func (p *legacyProfile) Apply(pod *corev1.Pod, containerName string, target runtime.Object) error {
+	style, err := getDebugStyle(pod, target)
+	if err != nil {
+		return fmt.Errorf("legacy profile: %w", err)
+	}
+
+	switch style {
+	case node:
+		mountRootPartition(pod, containerName)
+		useHostNamespaces(pod)
+
+	case podCopy:
+		p.keeps.removeLabels(pod)
```

Comment:
I think this is a behavioral change for legacyProfile because we were not removing as default and now we are.
I think, we can leave it as is and even keep flags don't change it. As it is a legacy profile.

Reply:
Do you mean copy pod applied legacyProfile should keep annotations, probes and initContainers and doesn’t  keep labels(it is not kept now as mentioned [here](https://github.com/kubernetes/kubernetes/pull/123149#discussion_r1596918676)) regardless of --keep-* flags?

Reply:
Yes that is true. Because it is a legacy profile anyway and we shouldn't change the current behavior, unless it is fully broken, etc.

Reply:
Thanks!
So should we add additional flags descriptions like "it does not work in legacyProfile" related [that](https://github.com/kubernetes/kubernetes/pull/123149#discussion_r1596976171)?

Reply:
Let's don't add it for now. I think, adding more text just for a a legacy profile seems to be unnecessary. Isn't the recommended path is using the other profiles other than `legacy`?

Reply:
We can say that if you want to manage these resources, please use the other recommended paths.

Reply:
> Let's don't add it for now. 

Ok, I've got it.

> Isn't the recommended path is using the other profiles other than legacy?

By the way, how many users recognize it recommended to use other profiles than legacy profile and the existence of debugging profile?

- The official document doesn't seem to mention debugging profiles.(It's mentioned only in [KEP](https://github.com/kubernetes/enhancements/blob/master/keps/sig-cli/1441-kubectl-debug/README.md#debugging-profiles) now as I know.)
- "kubectl debug --help" doesn't seem to show any recommended paths regarding profiles and show default profile is legacy.

So IMO, I think many users don't know the existence of debugging profile and recommendation.
Shouldn't we add any description about debugging profile somewhere?

In addition, it is said that in [KEP](https://github.com/kubernetes/enhancements/blob/master/keps/sig-cli/1441-kubectl-debug/README.md#default-profile-and-automation-selection):

> Default Profile and Automation Selection
In order to provide a seamless experience and encourage use of [PodSecurity](http://kep.k8s.io/2579), the "auto" profile will automatically choose a profile that's compatible with the current security profile by examining the pod-security.kubernetes.io/enforce annotation on the namespace and selecting the most permissive of "general", "baseline", and "restricted" that the controller will allow.
> 
> This will become the default behavior, but in order to maintain backwards compatibility the "legacy" profile will be the default profile until the 1.25 release. When --profile is not specified kubectl debug will print a warning about the upcoming change in behavior.

Though auto profile is WIP(#120687) now, shouldn't we print any warning or recommendation about profile as a first step?

These are outside the scope of this PR, but I was a little curious.

Reply:
I assume that `legacy` name is self explanatory but I see your comments and it all makes sense. We should reconsider what is the path for legacy profile.